.github/workflows/main.yml

```diff
@@ -74,18 +106,29 @@ jobs:
           # Changes the polling interval used by the file watcher
           CHOKIDAR_INTERVAL: 20
           CHOKIDAR_USEPOLLING: 1
-      - name: Get test coverage flags
+
+          # split tests across multiple machines
+          CI_NODE_INDEX: ${{ matrix.machine }}
+          CI_NODE_TOTAL: 3
         if: '${{ !steps.release-check.outputs.IS_RELEASE }}'
+      - name: Get test coverage flags
         id: test-coverage-flags
         run: |-
           os=${{ matrix.os }}
           node=${{ matrix.node-version }}
           echo "::set-output name=os::${os/-latest/}"
           echo "::set-output name=node::node_${node//[.*]/}"
         shell: bash
-      - uses: codecov/codecov-action@v2
         if: '${{ !steps.release-check.outputs.IS_RELEASE }}'
+      - uses: codecov/codecov-action@v2
         continue-on-error: true
         with:
           file: coverage/coverage-final.json
           flags: ${{ steps.test-coverage-flags.outputs.os }},${{ steps.test-coverage-flags.outputs.node }}
+        if: '${{ !steps.release-check.outputs.IS_RELEASE }}'
+  all:
```

Comment:
This is here strictly to make branch protection rules easier, to avoid changing those based on matrix naming